Content.Server/Medical/InsideCryoPodSystem.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Content.Server.Atmos.EntitySystems;
+using Content.Server.Medical.Components;
+using Content.Shared.Medical.Cryogenics;
+using Content.Shared.Standing;
+
+namespace Content.Server.Medical
+{
+    public sealed class InsideCryoPodSystem: EntitySystem
+    {
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            // Atmos overrides
+            SubscribeLocalEvent<InsideCryoPodComponent, InhaleLocationEvent>(OnInhaleLocation);
+            SubscribeLocalEvent<InsideCryoPodComponent, ExhaleLocationEvent>(OnExhaleLocation);
+            SubscribeLocalEvent<InsideCryoPodComponent, AtmosExposedGetAirEvent>(OnGetAir);
+
+            SubscribeLocalEvent<InsideCryoPodComponent, DownAttemptEvent>(HandleDown);
```

Comment:
Right now it's handled in CryoPodSystem, where this component is ensured on the patient entity when inserting him: https://github.com/space-wizards/space-station-14/pull/11349/files#diff-1bea9cdee7424b2ae962d2615216fc1313c504e2a64195a77b973d0c79949057R145

And it is removed when the patient is ejected: https://github.com/space-wizards/space-station-14/pull/11349/files#diff-1bea9cdee7424b2ae962d2615216fc1313c504e2a64195a77b973d0c79949057R177
The latter also happens if the cryo pod is destroyed or deconstructed (through the DestructionEventArgs listener), which I believe should cover all the cases. I assume a failsafe on InsideCryoPodComponent which removes itself from the entity if the parent container isn't a cryo pod anymore would be safer, I'll address that.

Reply:
Yeah so if any other piece of code ejects it rn it would leak so you should also handle container removals. Also I think we have a destructible trigger to dump container contents which is probably better rather than subscribing to destruction event.

Reply:
Done, through the EntGotRemovedFromContainerMessage event listener. Should be the correct way to handle this if I understood container events correctly. I also found the trigger you were talking about, although it seems to have no containers whitelist so now the machine components also get ejected, but i think fixing that is outside the scope of the PR since it happens with other machines too.